apps/vanilla/src/pages/exercise2/scripts/Player.ts

```diff
@@ -1,39 +1,35 @@
 import { DiceAccumulator } from './DiceAccumulator';
 import { Publisher } from './Publisher';
 import { WINNING_POINTS } from './const';
-import { TurnResult, ISubscriber, IsActiveMessage, IsWinnerMessage } from './types';
+import { TurnInfo, ISubscriber } from './types';
 
 /** Player's status. */
-export class Player implements ISubscriber<TurnResult> {
+export class Player implements ISubscriber<TurnInfo> {
 
 	/** Turns results. */
-	public readonly results: DiceAccumulator;
+	public readonly results = new DiceAccumulator();
 
 	/** Player is active. */
-	public readonly isActive: Publisher<IsActiveMessage>;
+	public readonly isActive = new Publisher();
 
 	/** Player is winner. */
-	public readonly isWinner: Publisher<IsWinnerMessage>;
+	public readonly isWinner = new Publisher();
```

Comment:
Plase don't forget about types

```suggestion
	public readonly isWinner = new Publisher<boolean>();
```